node.ts: implement Node.childElementCount

N8Brooks opened the pull request:
Implement Node.childElementCount.

The Document.childElementCount read-only property returns the number of child elements of the document.

https://developer.mozilla.org/en-US/docs/Web/API/Element/childElementCount
https://developer.mozilla.org/en-US/docs/Web/API/Document/childElementCount

N8Brooks reviewed (commented):
Removed repeated test

b-fuze commented on src/dom/node.ts:
```diff
@@ -321,6 +321,14 @@ export class Node extends EventTarget {
+  get childElementCount(): number {
+    let count = 0;
+    for (const { nodeType } of this.childNodes) {
+      count += +(nodeType === NodeType.ELEMENT_NODE);
```
Change this to an if statement please

b-fuze commented on test/units/Node-childElementCount.ts:
```diff
@@ -0,0 +1,24 @@
+  const document = new Document();
+  const divider = document.createElement("div");
+  divider.appendChild(document.createElement("p"));
+  divider.appendChild(document);
```
So... This is actually supposed to throw lol... You're not supposed to be able to append documents. What you can do instead is something like
```javascript
divider.appendChild(new Comment());
```
I'll need to push a fix for this later so it conforms to the **insert algorithm** step 7.1 "adopt node into parent's document" which should fail for actual documents as [specified here](https://dom.spec.whatwg.org/#concept-node-insert).

b-fuze reviewed (commented):
Thanks for the PR

N8Brooks commented:
Updated!

b-fuze commented:
Sorry, I just noticed that `childElementCount` is not implemented for `Node` and is instead implemented on `Element` and `Document`... It'd be great if you could move it to `element.ts` and `document.ts`, otherwise I can merge it and do it later.

N8Brooks commented:
Moved it to both `document.ts` and `element.ts`. Removed it from `node.ts`. Updated unit tests.

Also, added `.vscode` to `.gitignore` so it didn't pick up the `.vscode` directory created by the deno vscode extension. Seemed like there was a precedent for this because of the `.vim` in there.
